### PR TITLE
MBS-10104: Convert the area index page to React

### PR DESCRIPTION
**Pull request by reosarevok**
https://tickets.metabrainz.org/browse/MBS-10104

### CODE REVIEW

**Comment by mwiencek** [comment]
Whoops, didn't see the weird Flow failures before I approved. Force-pushed some weird fixes for those.

**Comment by mwiencek** [comment]
Just realized I made similar changes in https://github.com/metabrainz/musicbrainz-server/pull/1003 but I think I prefer the ones here. 🤔

**Comment by reosarevok** [comment]
Then maybe you want to edit that branch to make the same changes as here? ;)

File: root/area/AreaIndex.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  from '../static/scripts/common/components/WikipediaExtract';
+import Relationships from '../components/Relationships';
+import * as manifest from '../static/manifest';
+
+import AreaLayout from './AreaLayout';
+
+type Props = {|
+  +area: AreaT,
+  +numberOfRevisions: number,
+  +wikipediaExtract: WikipediaExtractT,
```
Comment (yvanzo) [inline root/area/AreaIndex.js]: Same as https://github.com/metabrainz/musicbrainz-server/pull/1003/files#r285043758
```suggestion
  +wikipediaExtract: WikipediaExtractT | null,
```

File: root/area/AreaIndex.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}: Props) => (
+  <AreaLayout entity={area} page="index">
+    <Annotation
+      annotation={area.latest_annotation}
+      collapse
+      entity={area}
+      numberOfRevisions={numberOfRevisions}
+    />
+    <WikipediaExtract
+      cachedWikipediaExtract={wikipediaExtract || null}
```
Comment (yvanzo) [inline root/area/AreaIndex.js]: Same as https://github.com/metabrainz/musicbrainz-server/pull/1003/files#r285047960

```suggestion
      cachedWikipediaExtract={wikipediaExtract}
```

File: root/static/scripts/common/components/Annotation.js [hunk 3]
```diff
@@ -19,17 +19,24 @@ import entityHref from '../utility/entityHref';
 import Collapsible from './Collapsible';
 import EditorLink from './EditorLink';
 
+type EntityProp = $ReadOnly<MinimalCoreEntityT & {
```
Comment (yvanzo) [inline root/static/scripts/common/components/Annotation.js]: Same as https://github.com/metabrainz/musicbrainz-server/pull/1003/files#r284997688

File: root/static/scripts/common/components/Annotation.js [hunk 4]
```diff
@@ -19,17 +19,24 @@ import entityHref from '../utility/entityHref';
...
+  +latest_annotation?: AnnotationT,
+}>;
+
 type Props = {|
   +$c: CatalystContextT | SanitizedCatalystContextT,
-  annotation: ?{...AnnotationT, editor: SanitizedEditorT},
+  +annotation: ?({+editor: EditorT | SanitizedEditorT | null} & AnnotationT),
```
Comment (yvanzo) [inline root/static/scripts/common/components/Annotation.js]: Same as https://github.com/metabrainz/musicbrainz-server/pull/1003/files#r284997961

**Pull request merged**